### PR TITLE
sql: make `jsonb_array_to_string_array` return NULL elements

### PR DESCRIPTION
**Pull request by maryliag**
This commit makes the builtin `jsonb_array_to_string_array` return NULL elements, which were previously being removed. It also adds more testing for different types of elements on the array.

Followup from https://github.com/cockroachdb/cockroach/pull/112865#pullrequestreview-1695142250

Epic: none

Release note (sql change): The newly added builting jsonb_array_to_string_array no longer removes NULL objects and includes them into the resulting array.

### CODE REVIEW

**Comment by maryliag** [comment]
friendly ping @mgartner

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112975#-:-Nhm03UOAxSPT3Qn8gzt:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @maryliag and @mgartner)

___
*[`-- commits` line 12 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/112975#-Nhm-hJG3NWIrodHOd17):*
nit: I'd reword this to something like "The newly added builting `jsonb_array_to_string_array` no longer removes NULL objects and includes them into the resulting array."

**Comment by maryliag** [comment]
TFTR!

bors r+

**Comment by mgartner** [comment]
Thanks for fixing this! Sorry I was so slow to review.
